File: src/commonMain/kotlin/me/kolterdyx/compiler/statement/BlockStatement.kt
package me.kolterdyx.compiler.statement

import me.kolterdyx.compiler.ast.StatementVisitor

class BlockStatement(
    val statements: List<Statement>
) : Statement() {
    override fun <R> accept(visitor: StatementVisitor<R>): R {
        return visitor.visitBlock(this)
    }
}

File: internal/ast/blockStatement.go
package ast

type BlockStmt struct {
	Statement

	Statements []Statement
}

func (b BlockStmt) Accept(visitor StatementVisitor) any {
	return visitor.VisitBlock(b)
}

func (b BlockStmt) Type() NodeType {
	return BlockStatement
}

func (b BlockStmt) ToString() string {
	body := ""
	for _, stmt := range b.Statements {
		body += stmt.ToString() + "\n"
	}
	return body
}
